rules/Php81/Rector/ClassMethod/NewInInitializerRector.php

```diff
@@ -112,6 +119,10 @@ public function refactor(Node $node): ?Node
                     continue;
                 }
 
+                if ($this->stmtsManipulator->isVariableUsedInNextStmt($stmts, $key + 1, $paramName)) {
```

Comment:
I'm not familiar with rector code, so what does `nextstmt` means ? @samsonasik 

Does it also fix the bug for code like 
```
public function __construct(?\DateTime $time = null)
    {
        $this->timeWasSet = null !== $time; // Here the statement is before 
        $this->time = $time ?? new \DateTime();
    }
```
or
```
public function __construct(?\DateTime $time = null)
    {
        $this->time = $time ?? new \DateTime();

        // Some code...

        $a = random_int(0, 1);
        if ($a === 1) {
            $this->timeWasSet = null !== $time; // Here it's not the direct next statement
        }
    }
```

Reply:
Yes, that should cover that, feel free to try latest dev-main